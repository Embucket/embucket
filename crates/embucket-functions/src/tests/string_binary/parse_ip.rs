use crate::test_query;

test_query!(
    parse_ipv4,
    "SELECT PARSE_IP('192.168.242.188/24', 'INET') AS ipv4",
    snapshot_path = "parse_ip"
);

test_query!(
    parse_ipv6,
    "SELECT PARSE_IP('fe80::20c:29ff:fe2c:429/64', 'INET') AS ipv6",
    snapshot_path = "parse_ip"
);

test_query!(
    parse_permissive_error,
    "SELECT PARSE_IP('not_an_ip', 'INET', 1) AS err",
    snapshot_path = "parse_ip"
);

test_query!(
    parse_permissive_error_int_nonzero,
    "SELECT PARSE_IP('not_an_ip', 'INET', 5) AS err",
    snapshot_path = "parse_ip"
);

test_query!(
    parse_permissive_error_float_nonzero,
    "SELECT PARSE_IP('not_an_ip', 'INET', 0.5) AS err",
    snapshot_path = "parse_ip"
);

test_query!(
    parse_non_permissive_error_zero,
    "SELECT PARSE_IP('not_an_ip', 'INET', 0) AS err",
    snapshot_path = "parse_ip"
);

test_query!(
    parse_non_permissive_error_float_rounds_to_zero,
    "SELECT PARSE_IP('not_an_ip', 'INET', 0.4) AS err",
    snapshot_path = "parse_ip"
);
