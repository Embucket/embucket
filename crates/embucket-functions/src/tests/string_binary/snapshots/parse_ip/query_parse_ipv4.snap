---
source: crates/embucket-functions/src/tests/string_binary/parse_ip.rs
description: "\"SELECT PARSE_IP('192.168.242.188/24', 'INET') AS ipv4\""
---
Ok(
    [
        "+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+",
        "| ipv4                                                                                                                                                                                                                       |",
        "+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+",
        "| {\"family\":4,\"host\":\"192.168.242.188\",\"ip_fields\":[3232297660,0,0,0],\"ip_type\":\"inet\",\"ipv4\":3232297660,\"ipv4_range_end\":3232297727,\"ipv4_range_start\":3232297472,\"netmask_prefix_length\":24,\"snowflake$type\":\"ip_address\"} |",
        "+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+",
    ],
)
